### PR TITLE
Document key kind restrictions for Secure Session

### PR DESCRIPTION
**Pull request by iamnotacake**
As Secure Session API allows passing any type of key pair, but only works with EC keys, it's better to mention that fact in docs and language guides.

### CODE REVIEW

File: themis/crypto-theory/cryptosystems/secure-session.md [hunk 1]
```diff
@@ -167,7 +167,8 @@ The only requirement for peer ID is that it cannot be empty
 and it must uniquely identify the public key used by a peer.
 
 **Keypairs** can be securely generated with Themis key generation functions.
-Secure Session supports both EC and RSA key pairs.
+Secure Session only supports EC keys. RSA support may be added in the future.
```
Comment (vixentael) [inline themis/crypto-theory/cryptosystems/secure-session.md:170]:
```suggestion
Secure Session only supports EC keys. RSA support is available per request only.
```

File: themis/crypto-theory/cryptosystems/secure-session.md [hunk 2]
```diff
@@ -167,7 +167,8 @@ The only requirement for peer ID is that it cannot be empty
...
+Please [contact us](mailto:dev@cossacklabs.com) if you need to use a different key type.
```
Comment (vixentael) [inline themis/crypto-theory/cryptosystems/secure-session.md:171]:
```suggestion
Please [contact us](mailto:dev@cossacklabs.com) if you need to use RSA or other key types.
```

File: themis/languages/cpp/features.md [hunk 3]
```diff
@@ -571,6 +571,10 @@ There you can find examples of Secure Session setup and usage in all modes.
 First, both parties have to generate [asymmetric keypairs](#asymmetric-keypairs)
 and exchange their public keys.
 The private keys should never be shared with anyone else.
+{{< hint info >}}
+**Note:**
+Secure Session only supports EC keys. RSA support may be added in the future.
```
Comment (vixentael) [inline themis/languages/cpp/features.md:576]: same as above

File: themis/crypto-theory/cryptosystems/secure-session.md [hunk 4]
```diff
@@ -178,12 +179,6 @@ themis_status_t themis_gen_ec_key_pair(
     size_t*  private_key_length,
     uint8_t* public_key,
     size_t*  public_key_length);
-
-themis_status_t themis_gen_rsa_key_pair(
-    uint8_t* private_key,
-    size_t*  private_key_length,
-    uint8_t* public_key,
-    size_t*  public_key_length);
```
Comment (ilammy) [inline themis/crypto-theory/cryptosystems/secure-session.md:186]: Nice attention to detail :)

**Pull request merged**